packages/menu/src/Menu.ts

```diff
@@ -201,16 +345,43 @@ export class Menu extends SpectrumElement {
             item = this.menuItems[index] as MenuItem;
         }
         index = Math.max(index, 0);
+        const selectedItemsMap = new Map<MenuItem, boolean>();
+        const selected: string[] = [];
+        const selectedItems: MenuItem[] = [];
         this.menuItems.forEach((item, i) => {
+            if (item.selected) {
+                selectedItemsMap.set(item, true);
+                selected.push(item.value);
+                selectedItems.push(item);
+            }
             if (i !== index) {
                 item.focused = false;
             }
         });
+        this.selectedItemsMap = selectedItemsMap;
+        this.selected = selected;
+        this.selectedItems = selectedItems;
         this.focusedItemIndex = index;
         this.focusInItemIndex = index;
     }
 
-    private prepItems = (): void => {
+    private async updateItemRoles() {
+        const childMenuItems = [
+            ...this.querySelectorAll(`sp-menu-item`),
+        ] as MenuItem[];
```

Comment:
What I'm seeing after initialization in the context of MenuGroup is that this is still an empty array in the last call. I'm guessing there's something I'm not quite grokking in the lifecycle, or that `MutationObserver` doesn't actually work for re-slotted children.

Reply:
`this.querySelectorAll` won't see re-slotted children. We'll need to move to something like `slot.assignedElements()` or the like.

Reply:
Perfect -- thanks!